pkg/arvo/app/ping.hoon

```diff
@@ -406,62 +154,33 @@
 ++  on-agent
   |=  [=wire =sign:agent:gall]
   ^-  [(list card) _this]
-  =^  cards  state
-    ?+    wire  `state
-        [%nat *]
-      ?.  ?=(%nat -.plan.state)  `state
-      ?.  ?=(%poke-ack -.sign)   `state
-      (take-ping:nat now.bowl wire p.sign)
-    ::
-        [%pub *]
-      ?.  ?=(%pub -.plan.state)  `state
-      ?.  ?=(%poke-ack -.sign)   `state
-      (take-pings:pub wire p.sign)
-    ::
-        [%one *]
-      ?.  ?=(%one -.plan.state)  `state
-      ?:  ?=(%poke-ack -.sign)   `state
-      :: XX handle error?
-      `state
-    ==
-  [cards this]
+  ?.  ?=([%ping s=@ *] wire)
+    `this
+  ?.  =(galaxy.state (slav %p i.t.wire))
+    `this
+  =.  sent.state  %.n
+  ?.  ?=(%formal mode.state)  `this
+  [[(wait-card now.bowl)]~ this]
```

Comment:
We need check that this is a `%poke-ack`, and explicitly handle nacks.

I think we could set the timer in %formal mode and when there's an error (`|(?=(%formal mode.state) ?=(^ p.sign.state))`). We might want to also switch to %formal mode on error -- less sure about that.

 If it's nack, we should either a) toggle the state to `%formal` or b) just retry once.